src/mlpack/tests/cf_test.cpp

```diff
@@ -256,14 +299,16 @@ BOOST_AUTO_TEST_CASE(CFPredictTest)
   totalError = std::sqrt(totalError) / savedCols.n_cols;
 
   // The mean squared error should be less than one.
-  BOOST_REQUIRE_LT(totalError, 0.5);
+  BOOST_REQUIRE_LT(totalError, 0.6);
```

Comment:
I don't think `0.6` is a good bound for the test here. If we use RMSE as metric, we would have
```
RMSE = totalError * sqrt(savedCols.n_cols) = totalError * sqrt(50)
````
Let's suppose `totalError = 0.6`, we can get `RMSE = 0.6 * sqrt(50) = 4.24`. For the dataset used in this test, rating range is `[1, 5]`, which means that, even with random prediction, RMSE still won't be larger than `4`.

Reply:
Nice catch! We should fix that.

Reply:
Should I comment on the CF refactoring PR, or do you like to do this inside this PR?

Reply:
@zoq I'd like to fix this in this PR

Reply:
great, thanks!